contracts/base/strategies/iron-fold/IronFoldStrategyBase.sol

```diff
@@ -0,0 +1,298 @@
+  /// @param _vault SmartVault address that will provide liquidity
+  /// @param __rewardTokens Reward tokens that the strategy will farm
+  /// @param _rToken RToken address
+  /// @param _ironController Iron Controller address
+  constructor(
+    address _controller,
+    address _underlying,
+    address _vault,
+    address[] memory __rewardTokens,
+    address _rToken,
+    address _ironController,
+    uint256 _borrowTargetFactorNumerator,
+    uint256 _collateralFactorNumerator,
+    uint256 _factorDenominator,
+    bool _fold
+  ) StrategyBase(_controller, _underlying, _vault, __rewardTokens, _BUY_BACK_RATIO) {
+    require(_rToken != address(0), "zero address pool");
+    require(_ironController != address(0), "zero address ironController");
+    rToken = _rToken;
+    ironController = _ironController;
+
+    address _lpt = CompleteRToken(rToken).underlying();
+    require(_lpt == _underlyingToken, "wrong underlying");
+
+    factorDenominator = _factorDenominator;
+    require(_collateralFactorNumerator < factorDenominator, "Collateral factor cannot be this high");
+    collateralFactorNumerator = _collateralFactorNumerator;
+    require(_borrowTargetFactorNumerator < collateralFactorNumerator, "Target should be lower than collateral limit");
+    borrowTargetFactorNumerator = _borrowTargetFactorNumerator;
+    fold = _fold;
+  }
+
+  // ************* VIEWS *******************
+
+  /// @notice Strategy balance supplied minus borrowed
+  /// @return bal Balance amount in underlying tokens
+  function rewardPoolBalance() public override view returns (uint256) {
+    return suppliedInUnderlying.sub(borrowedInUnderlying);
+  }
+
+  /// @notice Return approximately amount of reward tokens ready to claim in Iron MasterChef contract
+  /// @dev Don't use it in any internal logic, only for statistical purposes
+  /// @return Array with amounts ready to claim
+  function readyToClaim() external view override returns (uint256[] memory) {
+    uint256[] memory rewards = new uint256[](1);
+    return rewards;
+  }
+
+  /// @notice TVL of the underlying in the rToken contract
+  /// @dev Only for statistic
+  /// @return Pool TVL
+  function poolTotalAmount() external view override returns (uint256) {
+    return CompleteRToken(rToken).getCash().add(CompleteRToken(rToken).totalBorrows());
+  }
+
+  /// @notice Calculate approximately weekly reward amounts for each reward tokens
+  /// @dev Don't use it in any internal logic, only for statistical purposes
+  /// @return Array of weekly reward amounts
+  function poolWeeklyRewardsAmount() external view override returns (uint256[] memory) {
+    uint256[] memory rewards = new uint256[](1);
+    return rewards;
+  }
+
+  // ************ GOVERNANCE ACTIONS **************************
+
+  /// @notice Claim rewards from external project and send them to FeeRewardForwarder
+  function doHardWork() external onlyNotPausedInvesting override restricted {
+    claimReward();
+    liquidateReward();
```

Comment:
maybe call rebalance for each hard work?

Reply:
Good idea